BraintreeDropIn/BTDropInController.m

```diff
@@ -93,14 +93,18 @@ - (void)viewDidLoad {
 - (void)viewWillAppear:(BOOL)animated {
     [super viewWillAppear:animated];
 
-    if (self.isBeingPresented) {
-        [self.paymentSelectionViewController loadConfiguration];
-        [self resetDropInState];
-        [self loadConfiguration];
-    }
+    [self.paymentSelectionViewController loadConfiguration];
+    [self resetDropInState];
+    [self loadConfiguration];
     [self.apiClient sendAnalyticsEvent:@"ios.dropin2.appear"];
 }
 
+- (void)viewDidAppear:(BOOL)animated {
+    [super viewDidAppear:animated];
+    self.contentViewTopConstraint.constant = [self calculateContentViewTopConstraintConstant];
+    self.contentViewBottomConstraint.constant = -[self sheetInset];
```

Comment:
Hey do we adjust the constants because we're pinning our top and bottom constraints to the superview instead of the safe area layout guide?

Reply:
It's basically just a difference in how the view controller gets presented in SwiftUI vs. UIKit.

**UIKit**
`-[BTDropInControllerPresentTransition animateTransition:]` is called when the Drop-in comes on screen. That method calls `-[BTDropInController flexViewAnimated:]`, which has these lines in it:

```
self.contentViewTopConstraint.constant = [self calculateContentViewTopConstraintConstant];
self.contentViewBottomConstraint.constant = -[self sheetInset];
```

That's what ensures that the payment selection view controller (the little modal) is the correct height when it initially appears.

**SwiftUI**
`-[BTDropInControllerPresentTransition animateTransition:]` is never called. (If you want to animate the transition, you need to do it from SwiftUI.)

So in SwiftUI, we need to set `self.contentViewTopConstraint.constant` and `self.contentViewBottomConstraint.constant` somewhere else. We're using the safe area, which doesn't get calculated correctly until `viewDidAppear`, so that's when we set the constants on those two constraints.

Let me know if that makes sense!

Reply:
Yeah that makes sense. I forgot it was a pop-up modal for a second.